drivers/usb/udc/udc_dwc2.c

```diff
@@ -585,9 +585,18 @@ static void dwc2_handle_xfer_next(const struct device *dev,
 	if (USB_EP_DIR_IS_OUT(cfg->addr)) {
 		dwc2_prep_rx(dev, buf, cfg);
 	} else {
-		if (dwc2_tx_fifo_write(dev, cfg, buf)) {
-			LOG_ERR("Failed to start write to TX FIFO, ep 0x%02x",
-				cfg->addr);
+		int err = dwc2_tx_fifo_write(dev, cfg, buf);
+
+		if (err == -ENOENT) {
```

Comment:
Why there is arbitrary limit to `-ENOENT`? Why other errors are not treated the same?

Reply:
`-ENOENT` may not be related to an actual error (I observed it during USB cable disconnection if my HID device was providing HID reports over USB). Still it may be reasonable to call `udc_submit_ep_event` also for other errors here (I initially tried to limit scope my change to cases that I can test locally to avoid breaking something accidentally).